### PR TITLE
Migrate `local_test_autosplit.py` from pipeline driver to compile_stage

### PR DESCRIPTION
**Pull request by eddogola**
## Description

Since we're moving from the pipeline driver API to the compile_stage one, I make a corresponding update to a local test.

## Checklist:

- [x] Has code been commented, particularly in hard-to-understand areas?

### CODE REVIEW

**Comment by eddogola** [comment]
Find error here: https://www.internalfb.com/phabricator/paste/view/P792102156

**Comment by eddogola** [comment]
@kwen2501 @fegin @wz337

File: test/local_test_autosplit.py [hunk 1]
```diff
@@ -78,107 +67,131 @@ def inspect_split_module(
...
     with torch.autograd.profiler_legacy.profile(
         enabled=PROFILING_ENABLED
     ) as prof:
-        pipe_driver.chunks = nstages
-        out = pipe_driver(ec_input)
-        ref_out = ec_pipe(ec_input)
-        print(
-            f'profiling run completed {torch.sum(out["out"])} ref {torch.sum(ref_out["out"])}'
-        )
+        stage.chunks = nstages
```
Comment (kwen2501) [inline test/local_test_autosplit.py:104]: I don't think stage supports dynamically changing chunks

Comment (eddogola) [reply test/local_test_autosplit.py:104]: oh ok, will get rid of that

**Comment by eddogola** [comment]
@kwen2501 here's the code that breaks. See error here: https://www.internalfb.com/phabricator/paste/view/P797550529

**Comment by kwen2501** [comment]
Completed.

**Pull request closed without merging**